### ISSUE TITLE
Add @InternalApi where applicable

### ISSUE THREAD

**Issue by octonato** [akka/akka-persistence-jdbc#472]
We should be more clear what it internal API and what not. I will say that for a plugin, most APIs are internal. 

The DAOs and eventually some utility classes can be public. 

I suggest that we schedule this to be done before `5.0.0`.

**Comment by patriknw** [comment]
I don't think we should spend time on trying to make current shape of the plugin binary compatible with a well defined public api. The heavy usage of traits and no thought of that from the beginning makes it impossible. We will eventually rewrite it and then we should take this into account.
